### PR TITLE
Add support for Aurora Corne by splitkb

### PR DESCRIPTION
**Pull request by artursmirnov**
Mostly just added a keymap layout for the keyboard and enabled Vial support following the guide. Tested the integration with my keyboard.

### CODE REVIEW

File: keyboards/splitkb/aurora/corne/keymaps/vial/keymap.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+    "keyboard": "splitkb/aurora/corne/rev1",
+    "keymap": "default",
```
Comment (lesshonor) [inline keyboards/splitkb/aurora/corne/keymaps/vial/keymap.json:3]:
```suggestion
    "keymap": "vial",
```

File: keyboards/splitkb/aurora/corne/keymaps/vial/keymap.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            "RGB_MOD" , "RGB_HUD" , "RGB_SAD" , "RGB_VAD" , "XXXXXXX" , "XXXXXXX" , "XXXXXXX" , "XXXXXXX" , "XXXXXXX" , "XXXXXXX" , "XXXXXXX" , "XXXXXXX" ,
+                                                "KC_LGUI" , "_______" , "KC_SPC"  , "KC_ENT"  , "_______" , "KC_RALT"
+        ]
+
+    ],
+    "config": {
+        "features": {
+            "oled": true,
+            "rgb_matrix": true,
+            "rgblight": false
```
Comment (lesshonor) [inline keyboards/splitkb/aurora/corne/keymaps/vial/keymap.json:37]:
```suggestion
            "rgblight": false,
            "via": true,
            "vial": true
```

File: keyboards/splitkb/aurora/corne/keymaps/vial/rules.mk [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+VIA_ENABLE = yes
+VIAL_ENABLE = yes
```
Comment (lesshonor) [inline keyboards/splitkb/aurora/corne/keymaps/vial/rules.mk:17]: I'd say that you could delete this file entirely but you probably want to start selectively disabling features here so the firmware actually builds. Here's an example of that syntax.
```suggestion
ifeq ($(strip $(MCU)), atmega32u4)
    QMK_SETTINGS = no
endif
```
...Also you can leave the huge license blob off `rules.mk` files. QMK's collaborators [have actually been trying to get splitkb's contractors to stop cut-pasting it in](https://github.com/qmk/qmk_firmware/pull/21499#issuecomment-1632697365).

**Review by lesshonor — CHANGES_REQUESTED** [review]
> Tested the integration with my keyboard.

...I'm going to be charitable and assume that you're [using an ARM Micro](https://docs.qmk.fm/#/feature_converters?id=converters), because otherwise this keymap is *massively* oversize.

At present keymaps submitted to this repository still need to build for their original target platforms (i.e. AVR). The following resources may help:

- https://get.vial.today/docs/firmware-size.html
- https://docs.qmk.fm/#/squeezing_avr

**Comment by thomasbaart** [comment]
Hi! We're starting to add Vial support for all our keyboards soon, so we'll either raise changes to this PR or raise a new PR once we're ready. Thanks for the pointers @lesshonor, I'll make sure we take them into account.

Thank you Artur for the work! It's cool to see keymaps being added :)

The license headers already existed in other firmware that did get approved, so we tried to match as closely to existing firmware as possible. The process isn't always uniform for every PR due to the nature of the project, so I reckon things need a little more ironing sometimes ;)

Now I'm here anyway: Would it be worthwhile to raise all keyboards as a single PR, so we can apply any desired changes in one swoop and have it all nice and equal?

**Comment by lesshonor** [comment]
> Would it be worthwhile to raise all keyboards as a single PR, so we can apply any desired changes in one swoop and have it all nice and equal?

You can add and/or revise keymaps for all your boards in one fell swoop if you want. If everything passes CI checks it will almost certainly be merged (at xyz's convenience; I have no control over the process).

**Comment by xyzz** [comment]
Closing because there's been no update for over a month.

**Pull request closed without merging**